pkg/clusteragent/autoscaling/workload/telemetry.go

```diff
@@ -137,15 +151,45 @@ func setHorizontalScaleAppliedRecommendations(toReplicas float64, ns, targetName
 }
 
 func unsetHorizontalScaleAppliedRecommendations(ns, targetName, autoscalerName string) {
-	for _, source := range validRecommendationSources {
-		telemetryHorizontalScaleAppliedRecommendations.Delete(
-			ns,
-			targetName,
-			autoscalerName,
-			string(source),
-			le.JoinLeaderValue,
-		)
-	}
+	unsetHorizontalTelemetry(ns, targetName, autoscalerName, telemetryHorizontalScaleAppliedRecommendations)
```

Comment:
nit: seems like these functions are called once, would be a bit less verbose to have everything in the `deletePodAutoscalerTelemetry` with a single definition of each label set + delete on all related metrics.